maven-plugin/src/main/java/org/openmrs/maven/plugins/Deploy.java

```diff
@@ -151,13 +160,51 @@ private void runInteractiveMode(Server server, ServerUpgrader upgrader) throws M
                 upgrader.upgradePlatform(server, platform);
                 break;
             }
+            case(DEPLOY_OWA_OPTION):{
+                deployOwa(server, null, null);
+                break;
+            }
             default: {
                 throw new MojoExecutionException(String.format("Invalid installation option only '%s', '%s', '%s' are available",
-                        INSTALL_MODULE_OPTION, INSTALL_DISTRO_OPTION, INSTALL_PLATFORM_OPTION));
+                        DEPLOY_MODULE_OPTION, DEPLOY_DISTRO_OPTION, DEPLOY_PLATFORM_OPTION));
             }
         }
     }
 
+    private void deployOwa(Server server, String name, String version) throws MojoExecutionException {
+        OpenmrsBintray bintray = new OpenmrsBintray();
+        if(name == null){
+            List<String> owas = new ArrayList<>();
+            for(BintrayId id : bintray.getAvailableOWA()){
+                owas.add(id.getName());
+            }
+            name = wizard.promptForMissingValueWithOptions("Which OWA would you like to deploy?%s", name, "", owas, true);
+        }
+        if(version == null){
+            List<String> versions = bintray.getOwaMetadata(name).getVersions();
+            version = wizard.promptForMissingValueWithOptions("Which version would you like to deploy?%s", version, "", versions, true);
+        }
+        wizard.showMessage("Please remember that to run OWA you need to have openmrs-owa-module installed on the server");
+        File owaDir = new File(server.getServerDirectory(), "owa");
+        if(!owaDir.exists()){
+            //couldn't find place where owa dir is saved yet
+            boolean useDefaultDir = wizard.promptYesNo(String.format(
+                    "\nThere is no default directory '%s' on server %s, would you like to create it? (if not, you will be asked for path to custom directory)",
```

Comment:
Could you please explain this to me? Why would one want create a custom directory?